python/graphstorm/dataloading/dataloading.py

```diff
@@ -124,6 +124,8 @@ def __init__(self, sampler, dataset, construct_feat_ntype, construct_feat_fanout
         self._sampler = sampler
         self._construct_feat_sampler = _ReconstructedNeighborSampler(
                 dataset, construct_feat_ntype, construct_feat_fanout)
+        # Temporary fix for DGL 2.0.0+ Compatabilities
+        self.prob = None
```

Comment:
DGL fix this problem for 2.4.0, but as we mainly recommend users to use 2.3.0, better to use this one.